### DISCUSSION TITLE
How exactly does the build cache work?

### DISCUSSION THREAD

**Discussion by jonathanhuang13** [vercel/next.js#40588] [Help]
The [docs](https://nextjs.org/docs/advanced-features/ci-build-caching) are sparse and I'm wondering:

- What exactly is being cached in the build cache?
- Is it necessary to use the build cache (e.g. is the purpose just to speed up builds)?
- Is there a way to selectively say what shouldn't be cached?

The reason I'm asking is because I'm using Sentry and every time I build my app with the build cache, the Sentry SDK uses an old environment variable value. Whenever I redeploy without the cache, the new environment value is used. Here's the [issue](https://github.com/getsentry/sentry-javascript/issues/5734).

**Comment by gaston-flores** [comment]
Are you deploying on vercel? If so, you can look at [their documentation for build caching.](https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#understanding-build-cache). 

> What exactly is being cached in the build cache?

In the case of next, the files generated in `.next/cache/**`, plus your `node_modules/**`, `yarn.lock`, and `package-lock.json`

> Is it necessary to use the build cache (e.g. is the purpose just to speed up builds)?

It's only to speed up builds

> Is there a way to selectively say what shouldn't be cached?

I'm not sure to be honest, I think this depends on the framework preset.

**Reply by jonathanhuang13** [reply]
Thanks for the reply. I did see in the docs that `.next/cache/**`, node_modules and package-lock are cached, but what is in `.next/cache/**`? They're all pak files and I wanted to understand more of how those files get generated. Is Next.js using webpack caching under the hood? How does Next.js choose which files get cached?

**Reply by itse4elhaam** [reply]
@jonathanhuang13 Did you find something more detailed on this?

For just speeding up builds, it takes a massive chunk of space. Taking 500 odd MBs on my builds.

**Reply by jonathanhuang13** [reply]
Unfortunately, I didn't. My specific issue was fixed by manually setting the release in Sentry (see https://github.com/getsentry/sentry-javascript/issues/5734#issuecomment-1261483723)
